enhancements/microshift/microshift-updateability-ostree.md

```diff
@@ -0,0 +1,1156 @@
+         - `backups/current-commit.id/` exists
+           * Restore from `backups/current-commit.id/`
+
+         - `backups/current-commit.id/` does not exist
+           > Reminder: `prev-boot-commit` was unhealthy, `current-commit` was healthy and is missing backup
+           - `version.json` matches `current-commit.id`
+             > Means that `prev-boot-commit` possibly failed to make a backup. `version.json` is untouched so no migration attempt.
+             * Backup `backups/current-commit.id/`, continue running off `microshift/`
+           - `version.json` does not match `current-commit.id`
+             > Migrated without backup? Bug or user interference.
+             * Exit 1, block `microshift run`
+
+       - `current-commit.system == unhealthy`
+         > `prev-boot-commit` was staged over `unhealthy` `current-commit` without metadata cleanup.
+         > We do no support upgrade from unhealthy commit.
+         > Admin should've heal previous commit or delete MicroShift data before upgrade to start from clean state.
+         >
+         > This should be caught by *`prev-boot-commit.system` is `unhealthy` > `current-commit` != `prev-boot-commit` >
+         > `current-commit` does not exist in `health.json`*
+         * Exit 1, block `microshift run`
+
+   - `current-commit == prev-boot-commit`
+     > Because `current-commit` is already present in `health.json`, it's 2nd, 3rd,... boot of the commit (not 1st).
+
+     - `backups/current-commit.id/` exists
+       > commit was healthy at least once already.
+       > It means greenboot removed `boot_counter` and current boot isn't due to greenboot's automatic reboot.
+       > After addressing issues, admin should mark commit as healthy, so this shouldn't happen.
+       > Let's assume admin addressed the issues but forgot to mark as healthy and just rebooted the device.
+       * No backup, no restore, skip migration, allow `microshift run`
+         > If admin addressed issues, it should be healthy again.
+         > If issue persists, it'll end up unhealthy requiring manual intervention
+         > (which is the same result if we'd block `microshift run`).
+
+     - `backups/current-commit.id/` doesn't exist
+       > commit was always unhealthy - it wasn't "accepted" by greenboot, so it might still rollback.
+       > Because admin should heal the commit before manually rebooting, we assume it's reboot due to greenboot.
+
+       - `earlier-commit not in health.json`
+         > For MicroShift it's first commit it's running, but it doesn't mean it's the only commit
+         > (rollback commit might be without MicroShift).
+         >
+         > ostree status:
+         > 1. `current-commit` == `prev-boot-commit` - currently running,
+         >    last boot unhealthy, greenboot didn't accept it yet so it might still rollback
+         > 1. (optionally) rollback (possibly without MicroShift)
+         * Delete data, try from clean state
```

Comment:
I know this commit that this comment belongs to is no longer valid, but I think this is the tree:
```
1. `prev-boot-commit.system == unhealthy`
   - `current-commit == prev-boot-commit`
     - `backups/current-commit.id/` doesn't exist
       > commit was always unhealthy - it wasn't "accepted" by greenboot, so it might still rollback.
       - `earlier-commit not in health.json`
         > For MicroShift it's first commit it's running, but it doesn't mean it's the only commit
         > (rollback commit might be without MicroShift).
         >
         > ostree status:
         > 1. `current-deploy` == `prev-boot-deploy` - currently running,
         > 1. `current-commit` == `prev-boot-commit` - currently running,
         >    last boot unhealthy, greenboot didn't accept it yet so it might still rollback
         > 1. (optionally) rollback (possibly without MicroShift)
         * Delete data, try from clean state
```

Answering the question: it is the FDO workflow, but only until the first commit with microshift (`earlier-commit not in health.json` - from ushift perspective it's the first and only commit with microshift running)